locking/api.go

```diff
@@ -197,6 +197,12 @@ type committer struct {
 	Email string `json:"email"`
 }
 
-func newCommitter(name, email string) committer {
-	return committer{Name: name, Email: email}
+func NewCommitter(name, email string) Committer {
+	return Committer{Name: name, Email: email}
```

Comment:
Good question. The constructor function returns a raw-value `Committer`, since that's how it's embedded in the `Lock` type. `Lock` prefers a `Committer` over a `*Committer`, since the extra data needed to store the pointer of an immutable value is unnecessary.

However, there's no danger of mutable behavior happening in the `(*Committer) String() string` method, so copying the value is unnecessary. Go will automatically take the address to call methods defined on the pointer type, i.e. in a type defined like:

```go
type T struct {}

func (t *T) Foo() {}
```

calling

```go
t := T{}

t.Foo()
```

is equivalent to calling:

```go
t := T{}

(&t).Foo()
```